keep/posthog/posthog.py

```diff
@@ -1,16 +1,56 @@
 import os
-
+import time
+import uuid
 import posthog
+import logging
+import requests
+
+from importlib import metadata
 from posthog import Posthog
 
-if os.getenv("DISABLE_POSTHOG", "false") == "true":
+logger = logging.getLogger(__name__)
+
+DISABLE_POSTHOG = os.getenv("DISABLE_POSTHOG", "false") == "true"
+UPTIME_REPORTING_CADENCE = 60 * 60
+RANDOM_TENANT_ID_PERSISTENT_WITHIN_LAUNCH = uuid.uuid4()
+
+if DISABLE_POSTHOG:
     posthog.disabled = True
 
+try:
+    KEEP_VERSION = metadata.version("keep")
+except Exception:
+    KEEP_VERSION = os.environ.get("KEEP_VERSION", "unknown")
 
-def get_posthog_client():
+def get_posthog_client(sync_mode=False):
     posthog_api_key = (
         os.getenv("POSTHOG_API_KEY")
         or "phc_muk9qE3TfZsX3SZ9XxX52kCGJBclrjhkP9JxAQcm1PZ"
     )
-    posthog_client = Posthog(api_key=posthog_api_key, host="https://app.posthog.com")
+    posthog_client = Posthog(api_key=posthog_api_key, host="https://app.posthog.com", sync_mode=sync_mode)
     return posthog_client
+
+def is_posthog_reachable():
+    posthog_client = get_posthog_client(sync_mode=True)
+    try:
+        posthog_client.capture(
```

Comment:
what's the default timeout? can we put something here? I'm worried that in some air-gapped envs this will make startup "stuck"

Reply:
By default it's 3 sec, made explicit here: https://github.com/keephq/keep/pull/2366/files#diff-80e770595b2d185d5c97ed37847dd7566565fcf981ef6b25d2f88ba23642728bR34